libraries/tools/kotlin-gradle-plugin/src/main/kotlin/org/jetbrains/kotlin/gradle/targets/js/npm/NpmDependencyExtension.kt

```diff
@@ -19,82 +19,102 @@ interface NpmDependencyExtension {
     operator fun invoke(directory: File): NpmDependency
 }
 
-fun Project.addNpmDependencyExtension() {
-    val dependencies = this.dependencies as ExtensionAware
-
-    val npmDependencyExtension: NpmDependencyExtension = object : NpmDependencyExtension, Closure<NpmDependency>(dependencies) {
-        override operator fun invoke(name: String, version: String): NpmDependency =
-            NpmDependency(
-                project = this@addNpmDependencyExtension,
-                name = name,
-                version = version
-            )
+private abstract class AbstractNpmDependencyExtension(
+    dependencies: ExtensionAware,
```

Comment:
Seems that `dependencies` is not needed in constructor, it can been calculated from `project`